connectors/sources/mysql.py

```diff
@@ -106,11 +99,44 @@ async def close(self):
         await self.connection_pool.wait_closed()
         self.connection_pool = None
 
+    def _validate_connection_fields(self):
+        """Validates whether user input is empty or not for connection string
+
+        Raises:
+            Exception: Configured keys can't be empty
+        """
+        connection_fields = ["host", "port", "user", "password", "database"]
+        empty_connection_fields = []
+        for field in connection_fields:
+            if not self.configuration[field]:
+                empty_connection_fields.append(field)
+
+        if empty_connection_fields:
+            raise Exception(
+                f"Configured keys: {empty_connection_fields} can't be empty."
+            )
+
+        if (
+            isinstance(self.configuration["port"], str)
+            and not self.configuration["port"].isnumeric()
+        ):
+            raise Exception("Configured port can't include string literal.")
+
     async def ping(self):
         """Verify the connection with MySQL server"""
+        logger.info("Validating MySQL Configuration...")
+        self._validate_connection_fields()
+        connection_string = {
```

Comment:
since you are doing this validation just here, why not:

```
connection_string = self._build_connection_string()
```

and do the validation there?

Reply:
If we name method as _build_connection_string and validate config inside the method, it will conflict the function definition. So, it will be better if we can keep validation logic separate from building the connection string.
What do you think ?